diff --git a/makefile b/makefile
index e758e68..16a3b23 100755
--- a/makefile
+++ b/makefile
@@ -53,7 +53,7 @@ mips24kc_le_asm_aes: git_version
 #bcm2708:
 #	${cc_bcm2708} -o ${NAME}_bcm2708  -I. ${SOURCES} ${FLAGS} -lrt -static -O3
 amd64:git_version
-	${cc_local}   -o ${NAME}_$@    -I. ${SOURCES} ${FLAGS} -lrt -static -O3
+	${CXX}   -o ${NAME}_$@    -I. ${SOURCES} ${FLAGS} -lrt -static -O3
 
 amd64_perf:git_version
 	${cc_local}   -o ${NAME}_$@    -I. ${SOURCES} ${FLAGS} -lrt -static -O0 -fno-omit-frame-pointer -g
