diff --git a/power_assert.gemspec b/power_assert.gemspec
index ec7d4c2..5b48350 100644
--- a/power_assert.gemspec
+++ b/power_assert.gemspec
@@ -12,9 +12,7 @@ Gem::Specification.new do |s|
   s.summary     = "Power Assert for Ruby"
   s.description = "Power Assert shows each value of variables and method calls in the expression. It is useful for testing, providing which value wasn't correct when the condition is not satisfied."
 
-  s.files         = `git ls-files -z`.split("\x0").reject do |f|
-    f.match(%r{\A(?:test|spec|features|benchmark|bin)/})
-  end
+  s.files = [".github/workflows/ci.yml", ".gitignore", "BSDL", "COPYING", "Gemfile", "LEGAL", "README.md", "Rakefile", "lib/power_assert.rb", "lib/power_assert/colorize.rb", "lib/power_assert/configuration.rb", "lib/power_assert/context.rb", "lib/power_assert/enable_tracepoint_events.rb", "lib/power_assert/inspector.rb", "lib/power_assert/parser.rb", "lib/power_assert/version.rb", "power_assert.gemspec"]
   s.bindir        = 'exe'
   s.executables   = s.files.grep(%r{^exe/}) { |f| File.basename(f) }
   s.require_paths = ['lib']
