diff --git a/lua/blink/cmp/fuzzy/download/init.lua b/lua/blink/cmp/fuzzy/download/init.lua
index 47c7448..d8fa840 100644
--- a/lua/blink/cmp/fuzzy/download/init.lua
+++ b/lua/blink/cmp/fuzzy/download/init.lua
@@ -25,7 +25,7 @@ function download.ensure_downloaded(callback)
       -- no version file found, and found the shared rust library, user manually placed the .so file
       if version.current.missing and pcall(require, 'blink.cmp.fuzzy.rust') then return end
 
-      local target_git_tag = download_config.force_version or version.git.tag
+      local target_git_tag = download_config.force_version or 'flake'
 
       -- built locally
       if version.current.sha ~= nil then
