diff --git a/pcs/settings.py.in b/pcs/settings.py.in
index 6b80ea67..db5aeae7 100644
--- a/pcs/settings.py.in
+++ b/pcs/settings.py.in
@@ -9,7 +9,7 @@ pacemaker_binaries = "@PCMKEXECPREFIX@/sbin"
 corosync_binaries = "@COROEXECPREFIX@/sbin"
 corosync_qnet_binaries = "@COROQDEVEXECPREFIX@/bin"
 corosync_qdevice_binaries = "@COROQDEVEXECPREFIX@/sbin"
-corosync_conf_dir = "@COROCONFDIR@/corosync"
+corosync_conf_dir = "/etc/corosync"
 corosync_conf_file = os.path.join(corosync_conf_dir, "corosync.conf")
 corosync_uidgid_dir = os.path.join(corosync_conf_dir, "uidgid.d")
 corosync_qdevice_conf_dir = "@COROQDEVCONFDIR@"
