diff --git a/lua/plenary/popup/init.lua b/lua/plenary/popup/init.lua
index 92dec7f..8ebc5a5 100644
--- a/lua/plenary/popup/init.lua
+++ b/lua/plenary/popup/init.lua
@@ -190,6 +190,9 @@ function popup.create(what, vim_options)
   local win_opts = {}
   win_opts.relative = "editor"
   win_opts.style = "minimal"
+  if vim_options.border then
+    win_opts.border = 'rounded'
+  end
 
   -- Add positional and sizing config to win_opts
   add_position_config(win_opts, vim_options, { width = 1, height = 1 })
@@ -377,6 +380,7 @@ function popup.create(what, vim_options)
   end
 
   local border = nil
+  should_show_border = false
   if should_show_border then
     border_options.focusable = vim_options.border_focusable
     border_options.highlight = vim_options.borderhighlight and string.format("Normal:%s", vim_options.borderhighlight)
