diff --git a/lua/popup/init.lua b/lua/popup/init.lua
index e6ff845..36f8d27 100644
--- a/lua/popup/init.lua
+++ b/lua/popup/init.lua
@@ -151,6 +151,9 @@ function popup.create(what, vim_options)
   -- ,     contents on the screen.  Set to TRUE to disable this.
 
   win_opts.style = 'minimal'
+  if vim_options.border then
+    win_opts.border = 'rounded'
+  end
 
   -- Feels like maxheight, minheight, maxwidth, minwidth will all be related
   --
@@ -338,7 +341,7 @@ function popup.create(what, vim_options)
 
   local border = nil
   if should_show_border then
-    border = Border:new(bufnr, win_id, win_opts, border_options)
+    --border = Border:new(bufnr, win_id, win_opts, border_options)
   end
 
   if vim_options.highlight then
