From 39ec4ed59cff2982e7aba4392cfdc43d1fa40dca Mon Sep 17 00:00:00 2001
From: Viktor Kleen <viktor@kleen.org>
Date: Thu, 29 Apr 2021 21:36:21 +0000
Subject: [PATCH] adjust runtime path

---
 src/cli.cr | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cli.cr b/src/cli.cr
index 17427ee..5817e58 100644
--- a/src/cli.cr
+++ b/src/cli.cr
@@ -5,7 +5,7 @@ require "./kakoune"
 require "./env"
 
 PROGRAM_PATH = Path[Process.executable_path || PROGRAM_NAME]
-RUNTIME_PATH = PROGRAM_PATH.join("../../share/kcr").expand
+RUNTIME_PATH = PROGRAM_PATH.join("../share/kcr").expand
 
 module Kakoune::CLI
   extend self
-- 
2.31.1

