--- coreboot-4.7.orig/util/crossgcc/buildgcc	2018-01-14 16:44:43.000000000 -0800
+++ coreboot-4.7/util/crossgcc/buildgcc	2018-11-06 17:42:21.701110371 -0800
@@ -691,6 +691,7 @@
 		--target=${TARGETARCH} --enable-targets=${TARGETARCH}${ADDITIONALTARGET} \
 		--disable-werror --disable-nls --enable-lto --enable-gold \
 		--enable-interwork --enable-multilib \
+		--with-system-zlib \
 		--enable-plugins --enable-multilibs
 		CFLAGS="$HOSTCFLAGS" \
 		CXXFLAGS="$HOSTCFLAGS" \
